keps/sig-node/2570-memory-qos/README.md

```diff
@@ -104,6 +107,62 @@ Brief map as follows:
+| pod | sum(requests.memory) | n |
+| node | n/a | pods, kube-reserved, system-reserved | n |
+
+
+**Note:** The formula for memory.high is changed in Beta as the Alpha implementation has following problems:
+1. It fails to throttle when requested memory is closer to memory limits (or node allocatable): memory.high < requests.memory. e.g. If `requests.memory = 85, limits.memory=100`, and we have a throttling factor of 0.8, then as per the Alpha implementation memory.high =  memory throttling factor * limits.memory i.e. memory.high = 80. In this case the level at which throttling is supposed to occur i.e. memory.high is less than requests.memory. Hence there won't be any throttling as the Alpha implementation doesn't allow memory.high to be less than requested memory. 
+
+2. Default throttling factor of 0.8 might be too aggressive for some applications that are latency sensitive and always use memory close to memory limits.
+
+For detailed reasoning around the change, read the documentation [`here`](https://docs.google.com/document/d/1r-e_jWL5EllBgxtaNb9qI2eLOKf9wSuuuglTySWmfmQ/edit?usp=sharing).
+
+
+**Quality of Service for Pods**
+In addition to the change in formula for memory.high, we are also adding the support for memory.high to be set as per `Quality of Service(QOS) for Pod` classes. Following are the different cases:
+
+1. Guaranteed
+The pod gets a Guaranteed class if the request and limit values are the same. Guaranteed class pods are the highest priority pods and throttling is disabled for these pods by not setting memory.high thereby allowing these pods to use the maximum available/allocated memory.
+
+2. Burstable
+The pod gets a Burstable class if the requested memory is lower than the limit. 
+
+    Case I: When requests.memory and limits.memory are set, the forumula is used as-is: 
+    ```
+    memory.high = floor[ (requests.memory + memory throttling factor * (limits.memory - requests.memory)) / pageSize ] * pageSize
+    ```
+
+    Case II. When requests.memory is set, limits.memory is not set, we substitue limits.memory for node allocatable memory in the formula:
+    ```
+    memory.high = floor[ (requests.memory + memory throttling factor * (node allocatable memory - requests.memory))/ pageSize ] * pageSize
+    ```
+
+    Case III. When requests.memory is not set and  limits.memory is set, we set `requests.memory = 0` in the formula:
+    ```
+    memory.high = floor[ (memory throttling factor * limits.memory) / pageSize) ] * pageSize
+    ```
+
+3. BestEffort 
+The pod gets a BestEffort class if limits.memory and requests.memory are not set. We set `requests.memory = 0` and substitue limits.memory for node allocatable memory in the formula:
+    ```
+    memory.high = floor[ (memoryThrottlingFactor * node allocatable memory) / pageSize) * pageSize
+    ```
+
```

Comment:
nit: It might be worth to copy over from the doc a brief summary of the alternative discussed (i.e. annotation, etc) and describe why we decided to go with approach as described in the doc